### ISSUE TITLE
openAI response cost more than 30 seconds， timeout setting should be a configurable 

### ISSUE THREAD

**Issue by ufoe** [cogentapps/chat-with-gpt#137]
openAI response cost more than 30 seconds， timeout setting should be a configurable 。

server can set a more than 30 seconds timeout, users can set their own timeout limit

<img width="868" alt="image" src="https://user-images.githubusercontent.com/1306250/234500325-58f7b7f9-78d6-41e5-a9a4-d513d0a03c9b.png">

**Comment by PalAditya** [comment]
This seems like a property that can be defaulted to 30s and be allowed to set in https://github.com/cogentapps/chat-with-gpt/blob/main/server/src/config.ts. We can also read it and save it in the device, just like the openAPI key is read + saved. Happy to help implement it if this seems useful (and if I understood the process correctly 😅 )

**Comment by ufoe** [comment]
> This seems like a property that can be defaulted to 30s and be allowed to set in https://github.com/cogentapps/chat-with-gpt/blob/main/server/src/config.ts. We can also read it and save it in the device, just like the openAPI key is read + saved. Happy to help implement it if this seems useful (and if I understood the process correctly 😅 )

yes, it will be usefull if i can set the timeout  in config.yml like openai key

run it on docker 
 docker pull ghcr.io/cogentapps/chat-with-gpt:release
& docker run  

i can't change the timeout .

**Comment by ufoe** [comment]
btw 
i think it will be better to export the elevenlabs enable/disable setting in the config.yml.
